R/Prior.R

```diff
@@ -479,10 +495,73 @@ prior_invgamma <- function(alpha, beta, init = beta / (alpha - 1)) {
             "real<lower=0> prior_alpha_{name};",
             "real<lower=0> prior_beta_{name};"
         ),
-        init = init,
+        init = beta / (alpha - 1),
+        sample = \(n) local_rinvgamma(n, alpha, beta),
         validation = list(
             alpha = \(x) x > 0,
             beta = \(x) x > 0
         )
     )
 }
+
+
+
+
+
+#' Stub functions for sampling from distributions
+#'
+#' @description
+#' These functions only exist so that they can be mocked during unit
+#' tests in order to provide deterministic values. In most cases
+#' these are just straight forward pass throughs for the underlying
+#' distributions.
+#'
+#' @param alpha (`number`)\cr Parameter for underlying distribution.
+#' @param beta (`number`)\cr Parameter for underlying distribution.
+#' @param mu (`number`)\cr Parameter for underlying distribution.
+#' @param sigma (`number`)\cr Parameter for underlying distribution.
+#' @param nu (`number`)\cr Parameter for underlying distribution.
+#' @param ... Pass any additional arguments to the underlying distribution.
+#'
+#' @importFrom stats rbeta rcauchy rgamma rlnorm rlogis rnorm rt runif
+#'
+#' @name Local_Sample
+#' @keywords internal
+NULL
+
+#' @rdname Local_Sample
+local_rnorm <- \(...) rnorm(...)
+
+#' @rdname Local_Sample
+local_rcauchy <- \(...) rcauchy(...)
+
+#' @rdname Local_Sample
+local_rgamma <- \(...) rgamma(...)
+
+#' @rdname Local_Sample
+local_rlnorm <- \(...) rlnorm(...)
+
+#' @rdname Local_Sample
+local_rbeta <- \(...) rbeta(...)
+
+#' @rdname Local_Sample
+local_runif <- \(...) runif(...)
+
+#' @rdname Local_Sample
```

Comment:
I guess for these functions where there is some computation behind it, it would be good to document in a developer vignette where the formulas are coming from.